### PR TITLE
[sai-api-gen] Write files only when changes are detected

### PR DESCRIPTION
**Pull request by kcudnik**
This will make easier for make pick up when file actually changed

### CODE REVIEW

**Comment by kcudnik** [comment]
@chrispsommers please review

File: dash-pipeline/SAI/sai_api_gen.py [hunk 1]
```diff
@@ -357,6 +357,19 @@ def get_uniq_sai_api(sai_api):
     sai_api[TABLES_TAG] = tables
     return sai_api
 
+# don't write content to file if file alreay exists
```
Comment (chrispsommers) [inline dash-pipeline/SAI/sai_api_gen.py:360]: s/alreay/already/

**Review by chrispsommers — APPROVED** [review]
Minor typo, otherwise I approve (subject to CI passing). Thanks, great improvement!

File: dash-pipeline/SAI/sai_api_gen.py (hunk 1 above)
Comment (kcudnik) [reply dash-pipeline/SAI/sai_api_gen.py]: fixed

**Comment by kcudnik** [comment]
please merge if no more issues

**Comment by kcudnik** [comment]
btw. i added log lines when nothing to change locally:
```
nothing to change in ./SAI/experimental/saiexperimentaldashacl.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashacl.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashdirectionlookup.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashdirectionlookup.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashtag.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashtag.cpp
nothing to change in ./SAI/experimental/saiexperimentaldasheni.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidasheni.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashinboundrouting.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashinboundrouting.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashmeter.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashmeter.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashoutboundcatopa.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashoutboundcatopa.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashoutboundrouting.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashoutboundrouting.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashvnet.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashvnet.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashpavalidation.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashpavalidation.cpp
nothing to change in ./lib/sairoute.cpp
nothing to change in ./SAI/experimental/saiexperimentaldashvip.h
nothing to change in ./SAI/experimental/saiextensions.h
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./SAI/inc/saiobject.h
nothing to change in ./lib/saidashvip.cpp
nothing to change in ./SAI/experimental/saitypesextensions.h
nothing to change in ./lib/Makefile
nothing to change in ./lib/utils.cpp
nothing to change in ./lib/utils.h
nothing to change in ./lib/saifixedapis.cpp
nothing to change in ./lib/saiimpl.h
nothing to change in ./lib/logger.h
nothing to change in ./lib/logger.cpp
nothing to change in ./lib/saidash.h
nothing to change in ./lib/dashsai.h
nothing to change in ./lib/dashsai.cpp
nothing to change in ./lib/config.h
```

and generating script is not optimal ether, when sort and uniq:
```
       11 nothing to change in ./SAI/experimental/saiextensions.h
       12 nothing to change in ./SAI/experimental/saitypesextensions.h
       11 nothing to change in ./SAI/inc/saiobject.h
```

those 3 files are generated 11 and 12 times, didn't figure this out yet, but script is taking:
```
real    0m1.764s
user    0m1.737s
sys     0m0.017s
```

so potentially execution time could be also improved, maybe to like 0.5 sec etc, still p4 json needs to be parsed, but i can live with current time :D

**Comment by chrispsommers** [comment]
Hi @kcudnik for some reason https://github.com/sonic-net/DASH/pull/422 did not get merged last week as agreed, I just merged it, but you probably need to resync to main now.

**Comment by kcudnik** [comment]
there are no merge conflicts, jsut /azp run

**Pull request merged**